tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/amqp/connect/AMQPMirrorConnectionTest.java

```diff
@@ -243,9 +256,273 @@ public void testCreateDurableConsumerReplicatesAddressAndQueue() throws Exceptio
             consumer.close();
          }
 
+         peer.waitForScriptToComplete(500, TimeUnit.SECONDS);
+
+         server.stop();
+      }
+   }
+
+   @Test(timeout = 20000)
+   public void testBrokerMirrorHonorsCoreTunnelingEnable() throws Exception {
+      testBrokerMirrorHonorsCoreTunnelingEnableOrDisable(true);
+   }
+
+   @Test(timeout = 20000)
+   public void testBrokerMirrorHonorsCoreTunnelingDisable() throws Exception {
+      testBrokerMirrorHonorsCoreTunnelingEnableOrDisable(false);
+   }
+
+   public void testBrokerMirrorHonorsCoreTunnelingEnableOrDisable(boolean tunneling) throws Exception {
+      final Map<String, Object> brokerProperties = new HashMap<>();
+      brokerProperties.put(AMQPMirrorControllerSource.BROKER_ID.toString(), "Test-Broker");
+
+      final String[] capabilities;
+
+      if (tunneling) {
+         capabilities = new String[] {"amq.mirror", AmqpSupport.CORE_MESSAGE_TUNNELING_SUPPORT.toString()};
+      } else {
+         capabilities = new String[] {"amq.mirror"};
+      }
+
+      try (ProtonTestServer peer = new ProtonTestServer()) {
+         peer.expectSASLPlainConnect("user", "pass", "PLAIN", "ANONYMOUS");
+         peer.expectOpen().respond();
+         peer.expectBegin().respond();
+         peer.expectAttach().ofSender()
+                            .withName(Matchers.startsWith("$ACTIVEMQ_ARTEMIS_MIRROR"))
+                            .withDesiredCapabilities(capabilities)
+                            .respond()
+                            .withOfferedCapabilities(capabilities)
+                            .withPropertiesMap(brokerProperties);
+         peer.start();
+
+         final URI remoteURI = peer.getServerURI();
+         logger.info("Connect test started, peer listening on: {}", remoteURI);
+
+         AMQPMirrorBrokerConnectionElement mirrorElement = new AMQPMirrorBrokerConnectionElement();
+         mirrorElement.addProperty(TUNNEL_CORE_MESSAGES, Boolean.toString(tunneling));
+
+         AMQPBrokerConnectConfiguration amqpConnection =
+               new AMQPBrokerConnectConfiguration("testSimpleConnect", "tcp://" + remoteURI.getHost() + ":" + remoteURI.getPort());
+         amqpConnection.setReconnectAttempts(0);// No reconnects
+         amqpConnection.setUser("user");
+         amqpConnection.setPassword("pass");
+         amqpConnection.addElement(mirrorElement);
+         server.getConfiguration().addAMQPConnection(amqpConnection);
+         server.start();
+
          peer.waitForScriptToComplete(5, TimeUnit.SECONDS);
 
          server.stop();
       }
    }
+
+   @Test(timeout = 20000)
+   public void testProducerMessageIsMirroredWithNoCoreTunnelingUsesCoreMessageFormat() throws Exception {
+      doTestProducerMessageIsMirroredWithCorrectMessageFormat(true);
+   }
```

Comment:
Name needs tweaked to e.g remove No, since With**No**CoreTunnelingUsesCoreMessageFormat doesnt make sense.